pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -147,22 +147,17 @@ private String parseRemoteClusterName(String producerName, boolean isRemote) {
         return null;
     }
 
-    @Override
```

Comment:
Should we leave `hashCode()` & `equals()` with explanation on why it needs to be identity equal? Otherwise, someone will add it back later on...

Reply:
The tests I've added/updated should fail if the definition of equality changes or becomes too broad. If you are worried that the tests are insufficient, I'm happy to add the overrides back in. Let me know.